### PR TITLE
AV-197888 Additing validations to avoid attachment of httproute to Gateway in Invalid State

### PR DESCRIPTION
**Pull request by pkoshtavmware**
AV-197888 Additing validations to avoid attachment of httproute to Gateway in Invalid State

### CODE REVIEW

**Comment by pkoshtavmware** [comment]
build ako

**Comment by pkoshtavmware** [comment]
build ako

**Comment by pkoshtavmware** [comment]
build ako

**Comment by pkoshtavmware** [comment]
build ako

**Comment by pkoshtavmware** [comment]
build ako

**Comment by pkoshtavmware** [comment]
build ako

File: tests/gatewayapitests/status/httproute_test.go [hunk 1]
```diff
@@ -664,3 +664,125 @@ func TestHTTPRouteWithNoHostnames(t *testing.T) {
...
+	gatewayClassName := "gateway-class-hr-12"
+	gatewayName := "gateway-hr-12"
+	httpRouteName := "httproute-12"
+	namespace := "default"
+	ports := []int32{8080}
+
+	akogatewayapitests.SetupGatewayClass(t, gatewayClassName, akogatewayapilib.GatewayController)
+
+	listeners := akogatewayapitests.GetListenersV1(ports)
+	listeners[0].Hostname = nil
```
Comment (akshayhavile) [inline tests/gatewayapitests/status/httproute_test.go:735]: Are we setting up gateway to invalid by setting up hostname to nil? If that is the case, should AKO-GW accept such GW?

File: tests/gatewayapitests/status/httproute_test.go [hunk 2]
```diff
@@ -664,3 +664,125 @@ func TestHTTPRouteWithNoHostnames(t *testing.T) {
...
+	}, 30*time.Second).Should(gomega.Equal(true))
+
+	conditionMap := make(map[string][]metav1.Condition)
+
+	for _, port := range ports {
+		conditions := make([]metav1.Condition, 0, 1)
+		condition := metav1.Condition{
+			Type:    string(gatewayv1.GatewayConditionAccepted),
+			Reason:  string(gatewayv1.GatewayReasonInvalid),
+			Status:  metav1.ConditionFalse,
```
Comment (akshayhavile) [inline tests/gatewayapitests/status/httproute_test.go:706]: how are we making sure.. AKO-GW will not start processing GW object which is created at line number 685?

**Comment by pkoshtavmware** [comment]
build ako

**Comment by pkoshtavmware** [comment]
build ako

File: tests/gatewayapitests/status/httproute_test.go (hunk 2 above)
Comment (akshayhavile) [reply tests/gatewayapitests/status/httproute_test.go:706]: Can we spin up a go-routine with delay at beginning and then adding gateway to simulate the condition?

**Review by akshayhavile — APPROVED** [review]
Approved with one comment.

File: tests/gatewayapitests/status/httproute_test.go (hunk 2 above)
Comment (pkoshtavmware) [reply tests/gatewayapitests/status/httproute_test.go:706]: This is the only way to mimic this exact corner case. I have tried changing the testcase with both the above suggestions but in both of these cases parentref is [] and does not show this exact error
